### PR TITLE
fix: multi and single select questions

### PR DESCRIPTION
**Pull request by pandeymangg**
## What does this PR do?
Makes a common component for the `MultipleChoiceMulti` and `MultipleChoiceSingle` select questions called `SelectQuestion`. The types for both the questions types were also almost the same, only the logic had a difference, which is now being solved with `zod.refine`

Fixes # (issue)

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/types/surveys.ts [hunk 1]
```diff
@@ -296,29 +290,38 @@ export const ZSurveyConsentQuestion = ZSurveyQuestionBase.extend({
...
+  (question) => {
+    const { logic, type } = question;
+
+    if (type === TSurveyQuestionType.MultipleChoiceSingle) {
+      // logic ke andar 'includesAll'
+      return !logic?.some((l) => l.condition === "includesAll");
+    } else {
+      // logic ke andar 'notEquals'
+      return !logic?.some((l) => l.condition === "notEquals");
+    }
```
Comment (gupta-piyush19) [inline packages/types/surveys.ts:316]: please write comments in English only 😊

File: packages/database/data-migrations/20240318050527_add_languages_and_survey_languages/lib/i18n.ts [hunk 2]
```diff
@@ -160,21 +160,21 @@ const translateQuestion = (
...
       }
-      if (question.type === "multipleChoiceSingle") {
-        return ZSurveyMultipleChoiceSingleQuestion.parse(clonedQuestion);
-      } else return ZSurveyMultipleChoiceMultiQuestion.parse(clonedQuestion);
+
+      return ZSurveySelectQuestion.parse(clonedQuestion);
+
+    // if (question.type === "multipleChoiceSingle") {
+    //   return ZSurveyMultipleChoiceSingleQuestion.parse(clonedQuestion);
+    // } else return ZSurveyMultipleChoiceMultiQuestion.parse(clonedQuestion);
```
Comment (gupta-piyush19) [inline packages/database/data-migrations/20240318050527_add_languages_and_survey_languages/lib/i18n.ts:177]: please remove the comments.

File: apps/web/app/(app)/(survey-editor)/environments/[environmentId]/surveys/[surveyId]/edit/components/SelectQuestionForm.tsx [hunk 3]
```diff
@@ -318,7 +316,6 @@ export const MultipleChoiceSingleForm = ({
               variant="minimal"
               type="button"
               onClick={() => {
-                // @ts-expect-error
                 updateQuestion(questionIdx, { type: TSurveyQuestionType.MultipleChoiceMulti });
               }}>
               Convert to Multi Select
```
Comment (gupta-piyush19) [inline apps/web/app/(app)/(survey-editor)/environments/[environmentId]/surveys/[surveyId]/edit/components/SelectQuestionForm.tsx:321]: Please modify this button to change the type of the question(single to multi and vice versa).

**Review by gupta-piyush19 — CHANGES_REQUESTED** [review]
Thanks, @pandeymangg, for the changes 😊 
I've added a few comments 🙂 
Please make the changes accordingly 🙏

**Review by gupta-piyush19 — APPROVED** [review]

**Comment by gupta-piyush19** [comment]
LGTM 🚀 
Thanks, @pandeymangg  🙌

**Review by mattinannt — CHANGES_REQUESTED** [review]
@pandeymangg thanks for the PR :-) Looks good, the only thing I'm not so happy with is the naming. I would like to keep the term `MultipleChoice` instead of `Select` since we still have the two different question types `multipleChoiceSingle` & `multipleChoiceMulti`. Combining them in one form for me the name `MultipleChoiceQuestionForm` feels more consistent.

Please make the changes; then we are ready to merge this :-)

**Review by mattinannt — APPROVED** [review]

**Pull request merged**